art/attacks/inference/attribute_inference/meminf_based.py

```diff
@@ -148,9 +146,7 @@ def infer(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.n
         return pred_values
 
     def _check_params(self) -> None:
-        if not isinstance(self.attack_feature, int) and not isinstance(self.attack_feature, slice):
-            raise ValueError("Attack feature must be either an integer or a slice object.")
-        if isinstance(self.attack_feature, int) and self.attack_feature < 0:
-            raise ValueError("Attack feature index must be positive.")
+        self._check_attack_feature(self.attack_feature)
```

Comment:
Do all attribute inference attack have to call `self._check_attack_feature`? I'm wondering if we should generalize and autoamte it by moving it to `super._check_params`.